src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -12,6 +12,8 @@ namespace System.Text.Json
 {
     public sealed partial class JsonDocument
     {
+        private static readonly byte[] Utf8Bom = { 0xEF, 0xBB, 0xBF };
```

Comment:
```suggestion
        private static ReadOnlySpan<byte> Utf8Bom => new byte[] { 0xEF, 0xBB, 0xBF };
```
Its what all the cool kids are doing these days: https://vcsjones.com/2019/02/01/csharp-readonly-span-bytes-static/

Reply:
FWIW: This change (and some experimenting with extra locals) makes the synchronous ReadToEnd slightly longer assembly (0x459 + ret => 0x470 + ret); but shrank the async version (MoveNext 0x710 + ret => 0x659 + ret)

Interestingly, the smallest sync version came from caching the span to a local.  Caching span+length or length-only both resulted in even more assembly.